src/core/syscalls/business_logic_syscall_handler.rs

```diff
@@ -100,6 +162,65 @@ where
         Ok(syscall_request)
     }
 
+    fn syscall_deploy(
+        &mut self,
+        vm: &VirtualMachine,
+        syscall_request: SyscallRequest,
+        remaining_gas: u64,
+    ) -> Result<(Address, CallResult), SyscallHandlerError> {
+        let request = match syscall_request {
+            SyscallRequest::Deploy(request) => request,
+            _ => return Err(SyscallHandlerError::IncorrectSyscall("Deploy".to_string())),
```

Comment:
I think we could use an Error like the other we already have:
`SyscallHandlerError::ExpectedDeploySyscall` or `SyscallHandlerError::ExpectedDeployRequest`

Reply:
It is easier to use a generic one which can receive the name of the syscall, otherwise we will have to add an option for each syscall and i considerate it boiler plate.